### PR TITLE
Fixed!  Added a flatten step to EdgeQueryLogic.configureRanges to showcase te…

### PR DESCRIPTION
**Pull request by jwomeara**
…st errors in ExtendedEdgeQueryLogicTest and EdgeQueryFunctionalTest.  The test failures occur because EdgeTableRangeBuildingVisitor cannot handle a flattened tree.

This is broke... as a joke!

### CODE REVIEW

**Comment by jwomeara** [comment]
Spent a little time looking at this tonight.  I think I have it fixed now.  Let me know what you think.

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EdgeTableRangeBuildingVisitor.java [hunk 1]
```diff
@@ -168,75 +171,86 @@ public Object visit(ASTAndNode node, Object data) {
...
+                    mergedContext = childContext;
+                    continue;
+                } else if (((List<QueryContext>) mergedContext).get(0).getRowContext() != null) {
+                    for (QueryContext qContext : ((List<QueryContext>) mergedContext)) {
+                        if (!(qContext.combineQueryContexts(((List<QueryContext>) childContext), false))) {
+                            log.error("And node had unexpected return type");
+                            throw new IllegalArgumentException("Error: problem with query syntax");
+                        }
                     }
                 }
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EdgeTableRangeBuildingVisitor.java:243]: need a continue here

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EdgeTableRangeBuildingVisitor.java [hunk 2]
```diff
@@ -168,75 +171,86 @@ public Object visit(ASTAndNode node, Object data) {
...
-                    }
+            if ((childContext.get(0) instanceof IdentityContext) && (mergedContext.get(0) instanceof IdentityContext)) {
+                QueryContext qContext = new QueryContext();
+                
+                qContext.packageIdentities((List<IdentityContext>) childContext);
+                qContext.packageIdentities((List<IdentityContext>) mergedContext);
+                
+                ArrayList<QueryContext> aList = new ArrayList<>();
+                aList.add(qContext);
+                mergedContext = aList;
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EdgeTableRangeBuildingVisitor.java:200]: how about set "processed = true" as well and then test it at the error message below.

Comment (jwomeara) [reply warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EdgeTableRangeBuildingVisitor.java:200]: Check out my latest commit.  I think that by putting the thrown exception in an else statement we fix the issue and preserve the original functionality.

**Review by ivakegg — APPROVED** [review]

**Review by drewfarris — COMMENTED** [review]
The error message are the only potential issue that stuck out to me, but I'm not as familiar with this code as I could be.

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EdgeTableRangeBuildingVisitor.java [hunk 3]
```diff
@@ -168,75 +171,86 @@ public Object visit(ASTAndNode node, Object data) {
...
                     }
+                } else {
+                    log.error("Problem parsing query");
+                    throw new IllegalArgumentException("Error: problem with query syntax");
                 }
-                return contexts2;
+            } else {
+                
+                log.error("And node had unexpected return type");
+                throw new IllegalArgumentException("Error: problem with query syntax");
```
[the remark this answers is not in the document]
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EdgeTableRangeBuildingVisitor.java:249]: This is what was there before.....

**Pull request merged**